docs/internals/optimizer.rst

```diff
@@ -20,17 +22,21 @@ the function calls. Similarly, if a function is
 side-effect free and its result is multiplied by zero, you can remove the function
 call completely.
 
+The codegen-based optimizer currently consists of a single step that makes the compiler use unchecked
+arithmetic when generating code for the counter variable increment of certain ``for`` loops.
+
 Currently, the parameter ``--optimize`` activates the opcode-based optimizer for the
 generated bytecode and the Yul optimizer for the Yul code generated internally, for example for ABI coder v2.
 One can use ``solc --ir-optimized --optimize`` to produce an
 optimized Yul IR for a Solidity source. Similarly, one can use ``solc --strict-assembly --optimize``
 for a stand-alone Yul mode.
 
 .. note::
-    The `peephole optimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ is always
-    enabled by default and can only be turned off via the :ref:`Standard JSON <compiler-api>`.
+    The `peephole optimizer <https://en.wikipedia.org/wiki/Peephole_optimization>`_ and
+    the codegen-based optimizer are always enabled by default and can only be turned off
```

Comment:
I'd mention just the new step. `orderLiterals` is not always enabled and also when we add more codegen steps in the future, it will be easy to forget to update this note.

Reply:
Done.